File: ruby/lib/Sikaru/types/attachment_view_status.rb
# frozen_string_literal: true

module Sikaru
  module Types
    module AttachmentViewStatus
      extend Sikaru::Internal::Types::Enum

      PENDING = "pending"
      STARTING = "starting"
      READY = "ready"
      STOPPING = "stopping"
      CLEANED = "cleaned"
      STARTUP_EXPIRED = "startup_expired"
      RECOVERY_REQUIRED = "recovery_required"
      ABANDONED = "abandoned"
    end
  end
end

File: swift/Sources/Schemas/AttachmentViewStatus.swift
import Foundation

public enum AttachmentViewStatus: String, Codable, Hashable, CaseIterable, Sendable {
    case pending
    case starting
    case ready
    case stopping
    case cleaned
    case startupExpired = "startup_expired"
    case recoveryRequired = "recovery_required"
    case abandoned
}